### PR TITLE
Fields update

### PR DESCRIPTION
**Pull request by mickaelzhang**
## Screenshot / video
Screenshot 2022-09-01 at 11 42 26
Screenshot 2022-09-01 at 11 42 37

## What does this do?

- Add Fieldset on RadioGroup
- Share UI/logic for Field and Fieldset

### CODE REVIEW

File: src/fields/components/InternalField.tsx [hunk 1]
```diff
@@ -1,52 +1,59 @@
...
+interface InternalFieldProps extends InternalCommonFieldProps {
   children: ReactNode
+  className?: string
   assistiveText?: string
+  htmlFor?: string
+  fieldType?: 'field' | 'fieldset'
 }
 
-export const Field = ({
+export const InternalField = ({
```
Comment (mickaelzhang) [inline src/fields/components/InternalField.tsx:16]: Created an internal component that can be used on both `Field` and `Fieldset` so now they share the same UI and logic

File: src/fields/components/Placeholder.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import { Box } from '../../Box'
+import { theme } from '../../theme'
+
+export const Placeholder = () => {
```
Comment (mickaelzhang) [inline src/fields/components/Placeholder.tsx:7]: Screenshot 2022-09-01 at 11 34 00

Placeholder component for stories

File: src/fields/components/InternalField.tsx [hunk 3]
```diff
@@ -70,9 +77,12 @@ export const Field = ({
...
 
-const Container = styled(Box)<{ className: string }>`
+const Container = styled(Box)`
   display: flex;
   flex-direction: column;
   position: relative;
   width: 100%;
+
+  // In case, the element is a 'fieldset', we remove the border
+  border: 0;
```
Comment (mickaelzhang) [inline src/fields/components/InternalField.tsx:87]: @Jmuskett This should fix the issue

File: src/fields/components/InternalField.tsx [hunk 4]
```diff
@@ -1,52 +1,59 @@
...
+      className={className}
+      {...marginProps}
+    >
       {label && (
         <>
           {renderAsTitle ? (
             <Box mb="16px">
               <Text
-                tag="label"
+                tag={fieldType === 'field' ? 'label' : 'legend'}
```
Comment (mickaelzhang) [inline src/fields/components/InternalField.tsx:41]: Only difference between the 2 are the tags rendered for a11y purpose

File: src/fields/commonFieldTypes.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  label?: string
+  renderAsTitle?: boolean
+  assistiveText?: string
+  required?: boolean
+  error?: boolean
+  errorMsg?: string
+}
+
+export interface InternalCommonFieldProps extends CommonFieldProps {
+  outlined?: boolean
```
Comment (mickaelzhang) [inline src/fields/commonFieldTypes.ts:17]: `outlined` isn't needed on all component so I've moved it

File: src/NumberInput/NumberInput.tsx [hunk 6]
```diff
@@ -10,12 +10,12 @@ import styled from 'styled-components'
...
 import { Icon } from '../Icon'
-import { Field, CommonFieldTypes } from '../Field'
+import { Field, CommonFieldProps } from '../fields/Field'
 
 import { theme } from '../theme'
 import { useUniqueId } from '../utils/id'
 
-export interface Props extends CommonFieldTypes {
+export interface Props extends CommonFieldProps {
   type?: 'number' | 'tel'
```
Comment (githab666) [inline src/NumberInput/NumberInput.tsx:19]: Not sure `'number' and 'tel'` are very clear, could we change to something more helpful?

File: src/fields/components/InternalField.tsx (hunk 1 above)
Comment (githab666) [reply src/fields/components/InternalField.tsx:16]: Love this

**Review by githab666 — APPROVED** [review]
Minor comments - looking good :shipit:

File: src/NumberInput/NumberInput.tsx (hunk 6 above)
Comment (mickaelzhang) [reply src/NumberInput/NumberInput.tsx:19]: This is part of the API for `NumberInput` so changing might be breaking some stuff
Better keep it as it is for now but might be worth to have a look later on 👍

**Review by Jmuskett — APPROVED** [review]

**Pull request merged**